index.js

```diff
@@ -10,25 +10,23 @@ function generate(argIndex) {
 		var promise = thunk(value);
 		thunk.then = promise.then.bind(promise);
 		thunk.catch = promise.catch.bind(promise);
-		return thunk;
 
-		function thunk(result) {
-			if (promise) {
-				// Prevent unhandled rejection errors if promise is never used, and only used as a thunk
-				promise.catch(function () {});
-			}
+		// Prevent unhandled rejection errors if promise is never used, and only used as a thunk
+		promise.catch(function () {});
```

Comment:
You are right about that but now I'm realizing that the previous code didn't work as intended in the first place.

From what I understand `unhandledrejection` will be dispatched pretty soon after the error happens and that `.catch` call won't happen until `thunk` is called again, which in this case will happen 5000ms after the promise was rejected (and `unhandledrejection` event dispatched)

```js
doSomethingFor5000ms()
.then(delay.reject(3000))
```